firefox-ios/firefox-ios-tests/Tests/ClientTests/Frontend/Browser/TabScrollHandlerTests.swift

```diff
@@ -243,18 +233,83 @@ final class TabScrollHandlerTests: XCTestCase {
         XCTAssertEqual(delegate.hideCount, 0)
     }
 
+    func test_noScrollableContent_preventsUIUpdates() {
+        let smallContentSize = CGSize(width: 320, height: 500)
+        let subject = createSubject(contentSize: smallContentSize)
+
+        subject.handleScroll(for: CGPoint(x: 0, y: -80))
+        XCTAssertTrue(delegate.updateCalls.isEmpty)
+
+        subject.handleEndScrolling(for: CGPoint(x: 0, y: -80), velocity: .zero)
+        XCTAssertEqual(delegate.showCount, 0)
+        XCTAssertEqual(delegate.hideCount, 0)
+    }
+
+    // MARK: - endDrag checks
+
+    func test_endDrag_atBottom_preventsCommit() {
+        let contentOffset = CGPoint(x: 0, y: 1980)
+        let subject = createSubject()
+        tabProvider.scrollView?.contentOffset = contentOffset
+
+        // Simulate end drag; handler checks scrollReachBottom()
+        subject.scrollViewDidEndDragging(tabProvider.scrollView!, willDecelerate: false)
+
+        // No show/hide because at bottom
+        XCTAssertEqual(delegate.showCount, 0)
+        XCTAssertEqual(delegate.hideCount, 0)
+    }
+
+    func test_IgnoreScroll_tabIsLoading_preventsCommit() {
```

Comment:
```suggestion
    func test_ignoreScroll_tabIsLoading_preventsCommit() {
```

maybe we revisit naming eventually? Can add it to our next agenda!